### PR TITLE
HSQLDB integration

### PR DESCRIPTION
**Pull request by AtaydeEnrique**
## Description

HSQLDB (HyperSQL) Implementation 🚀

**Fixes** #4590 

## Type of change
- [x] ⚡ New feature (non-breaking change which adds functionality)

### Implementation

This handler was implemented using [pyodbc](https://pypi.org/project/pyodbc/), the Python ODBC bridge and the [Postgres ODBC Driver](https://www.postgresql.org/ftp/odbc/versions/). HyperSQL 

### Why was Postgres ODBC driver used?
From the [HyperSQL documentation](http://hsqldb.org/doc/2.0/guide/guide.html#odbc-chapt):
> Improvements were made to the server code for version 2.5.1 to allow an unmodified PostgreSQL ODBC driver (version 11) to be used. 

So basically, this implementation can be used to update the PostgreSQL implementation as well.

For more information refer to the readme file added in this pull request.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, or created issues to update them.
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

**Comment by AtaydeEnrique** [comment]
Forced a push since I edited one file I shouldn't have touched while copying and pasting for formatting the files in the handler I implemented.

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by AtaydeEnrique** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/integrations/handlers/hsqldb_handler/hsqldb_handler.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        """
+        Receive query as AST (abstract syntax tree) and act upon it somehow.
+        Args:
+            query (ASTNode): sql query represented as AST. May be any kind
+                of query: SELECT, INTSERT, DELETE, etc
+        Returns:
+            HandlerResponse
+        """
+
+        renderer = SqlalchemyRender('postgres')
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/hsqldb_handler/hsqldb_handler.py:151]: Is hsqldb postgresql compatible?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/hsqldb_handler/hsqldb_handler.py (hunk 1 above)
Comment (AtaydeEnrique) [reply mindsdb/integrations/handlers/hsqldb_handler/hsqldb_handler.py:151]: Hello, yes it is compatible as stated in the official page, you can read more about it here in the [compatibility page](https://hsqldb.org/doc/2.0/guide/compatibility-chapt.html) as well as in the [HyperSQL ODBC documentation](http://hsqldb.org/doc/2.0/guide/guide.html#odbc-chapt) in the "Overview" section of the page

I used HSQL database manager to create a database and connect it to MindsDB. The default demo database works too with this handler since it uses the Postgres driver.

image
image

**Review by ZoranPandovski — APPROVED** [review]
Thanks

**Comment by ZoranPandovski** [comment]
Great job @AtaydeEnrique feel free to check other integrations issues or post your own :raised_hands:

**Comment by AtaydeEnrique** [comment]
> Great job @AtaydeEnrique feel free to check other integrations issues or post your own 🙌

Awesome, for sure I will check them out! It's been challenging but I've been learning a lot. Thanks 💪🏼
